test/benchmarks/reduce/bench_thrust_reduce.cpp

```diff
@@ -0,0 +1,63 @@
+/* Copyright 2021 Simeon Ehrig
+ *
+ * This file is part of vikunja.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <numeric>
+#include <vector>
+
+#include <catch2/catch.hpp>
+#include <thrust/device_vector.h>
+
+template<typename TData>
+inline void benchmark_reduce(int size)
+{
+    std::vector<TData> hostMemInput(size);
+    for(int i = 0; i < size; ++i)
+    {
+        hostMemInput[i] = static_cast<TData>(i) + static_cast<TData>(1);
+    }
+
+    thrust::device_vector<TData> devMemInput(hostMemInput);
+
+    TData result = thrust::reduce(devMemInput.begin(), devMemInput.end(), static_cast<TData>(0));
+
+    TData expected_result = (static_cast<TData>(size) * (static_cast<TData>(size) + 1)) / static_cast<TData>(2);
+    // verify, that vikunja reduce is working with problem size
+    REQUIRE(expected_result == Approx(result));
+
+    // honeypot to check that the function call in the benchmark block has not been removed by the optimizer
+    result = static_cast<TData>(0);
+
+    BENCHMARK("reduce thrust")
+    {
+        result = thrust::reduce(devMemInput.begin(), devMemInput.end(), static_cast<TData>(0));
+    };
```

Comment:
According to the [Catch2 benchmark docs](https://github.com/catchorg/Catch2/blob/devel/docs/benchmarks.md) you could return `result`:
```suggestion
    BENCHMARK("reduce thrust")
    {
        return result = thrust::reduce(devMemInput.begin(), devMemInput.end(), static_cast<TData>(0));
    };
```
Catch2 will ensure that the value of `result` is used so the computation cannot be removed by the optimizer.

Reply:
I know about the problem but I thought it is only necessary for functions, which does not return a value and has no arguments or only arguments copied by value. Nevertheless, I can add it. 

By the way, that the reason, why I change the result after the first run and before the benchmark and check it afterwards again. That's show me, that the function was executed.

Reply:
Mind that Catch2 can decide to run the benchmark multiple times and the compiler could decide to eliminate that loop and only execute it once, since there is no side effect if you just write to `result`.

Reply:
Okay, that could be possible and it is not caught by my test. I will change it.